phase/extender_test.go

```diff
@@ -196,305 +196,622 @@ func testExtender(t *testing.T, when spec.G, it spec.S) {
+					}
+					h.Mkdir(t, filepath.Join(generatedDir, "run", "A"))
+					h.Mkfile(t, "some dockerfile content", filepath.Join(generatedDir, "run", "A", "Dockerfile"))
+					buf := new(bytes.Buffer)
+					data := extend.Config{Run: extend.BuildConfig{Args: expectedDockerfileA.Args}}
+					h.AssertNil(t, toml.NewEncoder(buf).Encode(data))
+					h.Mkfile(t, buf.String(), filepath.Join(generatedDir, "run", "A", "extend-config.toml"))
+
+					fakeDockerfileApplier.EXPECT().ImageFor(extender.ImageRef).Return(someFakeImage, nil)
+					firstConfig := &v1.ConfigFile{Config: v1.Config{
+						User: "0:5678",
+					}}
+					someFakeImage.ConfigFileReturns(firstConfig, nil)
+					someFakeImage.ManifestReturns(&v1.Manifest{Layers: []v1.Descriptor{}}, nil)
+
+					fakeDockerfileApplier.EXPECT().Apply(
+						gomock.Any(),
+						gomock.Any(),
+						gomock.Any(),
+						logger,
+					).DoAndReturn(
+						func(dockerfile extend.Dockerfile, _ v1.Image, _ extend.Options, _ llog.Logger) (v1.Image, error) {
+							h.AssertEq(t, dockerfile.Path, expectedDockerfileA.Path)
+							h.AssertEq(t, len(dockerfile.Args), 4)
+							h.AssertEq(t, dockerfile.Args[0].Name, "build_id")
+							_, err := uuid.Parse(dockerfile.Args[0].Value)
+							h.AssertNil(t, err)
+							h.AssertEq(t, dockerfile.Args[1].Name, "user_id")
+							h.AssertEq(t, dockerfile.Args[1].Value, "0")
+							h.AssertEq(t, dockerfile.Args[2].Name, "group_id")
+							h.AssertEq(t, dockerfile.Args[2].Value, "5678")
+							h.AssertEq(t, dockerfile.Args[3], expectedDockerfileA.Args[0])
+
+							return someFakeImage, nil
+						})
+
+					err := extender.Extend("run", logger)
+					h.AssertError(t, err, "extending run image: the final user ID is 0 (root); please add another extension that resets the user to non-root")
+				})
 			})
 		})
 
-		when("run base image", func() {
-			type testCase struct {
-				firstDockerfileRebasable  bool
-				secondDockerfileRebasable bool
-				expectedImageSHA          string
-			}
-			var (
-				rebasableSHA    = "sha256:2407b98f3bcee33d24f47fc3d7beaf0928b3fa799a8a84aa9f9f239fcded6e62"
-				notRebasableSHA = "sha256:cc3b90b798b76e7d41de01d4f02f2917694ef0b09d0d923314c8d6c31c4ae8e9"
-			)
-			for _, tc := range []testCase{
-				{
-					firstDockerfileRebasable:  true,
-					secondDockerfileRebasable: true,
-					expectedImageSHA:          rebasableSHA,
-				},
-				{
-					firstDockerfileRebasable:  true,
-					secondDockerfileRebasable: false,
-					expectedImageSHA:          notRebasableSHA,
-				},
-				{
-					firstDockerfileRebasable:  false,
-					secondDockerfileRebasable: true,
-					expectedImageSHA:          notRebasableSHA,
-				},
-				{
-					firstDockerfileRebasable:  false,
-					secondDockerfileRebasable: false,
-					expectedImageSHA:          notRebasableSHA,
-				},
-			} {
-				tc := tc
-				when := when
-				desc := func(b bool) string {
-					if b {
-						return "rebasable"
+		when("using platform API 0.13", func() {
```

Comment:
Similar comment to https://github.com/buildpacks/lifecycle/pull/1276#discussion_r1489634499, I wonder if we could just make vars for "A build.Dockerfile path", "A build BuildContext" (etc) and make this more of a table test.